src/scripts/main.js

```diff
@@ -1,5 +1,25 @@
 'use strict';
 
-document.addEventListener('click', e => {
-  // write code here
+document.addEventListener('click', el => {
+  const spider = document.querySelector('.spider');
+  const box = el.target;
+  let x = el.offsetX - spider.clientWidth / 2;
+  let y = el.offsetY - spider.clientHeight / 2;
+
+  if (x < 0) {
+    x = 0;
+  } else if (el.offsetX > box.clientWidth) {
+    x = box.clientWidth - spider.clientWidth;
+  }
+
+  if (y < 0) {
+    y = 0;
+  } else if (el.offsetY > box.clientHeight) {
+    y = box.clientHeight - spider.clientHeight;
+  }
+
+  if (box.classList.value === 'wall') {
```

Comment:
```suggestion
  if (box.classList.contains('wall')) {
```

Reply:
if you have 2 classes you'll get false here, so lets rework
also i propose you to move this condition on top of function and if box hasn't class 'wall' => return;